proposals/0010-stream-type.md

```diff
@@ -0,0 +1,162 @@
+
+\- From: *§5.3.1.2 Table 3 - Semantics of MPD element*
+
+### Recommended inferred value from native `HTMLMediaElement`
+
+```js
+streamType = Number.isFinite(mediaEl.duration)
+  ? "on-demand"
+  : mediaEl.duration === Number.POSITIVE_INFINITY
+  ? "live"
+  : "unknown"
+```
+
+__Context__:
+
+`media.duration`: 
+> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Returns the length of the media resource, in seconds, assuming that the start of the media resource is at time zero. <br/>
+> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Returns NaN if the duration isn't available. <br/>
+> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Returns Infinity for unbounded streams.
+
+\- From: https://html.spec.whatwg.org/multipage/media.html#dom-media-duration-dev
+
+
+## Event: `"streamtypechange"`
+
+This event should be dispatched from an extended `HTMLMediaElement` instance whenever the stream type has been determined for a loaded media source.
+
+__NOTE__: For media unload cases, implementors *__SHOULD__* ensure the `streamType` is (re)set to `"unknown"` after an unload begins but prior to any `"emptied"` external event handlers being invoked. This ensures that consumers of the API may reliably use the `"emptied"` event to monitor potential changes in `"streamType"`.
+
+# Rationale and alternatives
+
+This proposed solution provides a simple API with simple values for exposing information that is useful for UI development that distinguishes between live and on demand content, but is only guaranteed to be available by the code/entity responsible for parsing the media data. Based on the scope of currently proposed stream types, an alternative implementation could potentially rely on `duration` values (corresponding to the values used in *§Recommended inferred value from native `HTMLMediaElement`*, above). Below is a discussion on why this is not recommended.
+
+## Why not simply rely on `duration`?
+
+While there are elements of various specifications that suggest live content *should*, under most circumstances, have a `duration=Infinity` (See, e.g. https://html.spec.whatwg.org/multipage/media.html#dom-media-duration-dev, https://www.w3.org/TR/media-source-2/#dom-mediasource-setliveseekablerange, https://www.w3.org/TR/media-source-2/#dfn-duration-change), there are a few reasons to avoid this.
+
+1. Although https://html.spec.whatwg.org/multipage/media.html#dom-media-duration-dev strongly suggests that `duration` *__SHOULD__* be `Infinity` for live streams, there is at least some ambiguity between "live" vs. "unbounded" (referencing the language from the specification), so conflating the two may be inappropriate.
+2. At least some "playback engines" *__MAY__* set a finite duration for live streams, at least by default. While I do not know of any cases, this may also be true for some native playback environments.
```

Comment:
For web based players, I think it mostly happens for MSE playback that don't use the new API for setting seekable in MSE streams.